crossfire/__init__.py

```diff
@@ -51,16 +57,24 @@ def occurrences(
     )
 
 
+def _make_flatten(data, keys):
+    for item in data:
+        for key in keys:
+            item.update({f"{key}_{k}": v for k, v in item.get(key).items()})
+            item.pop(key)
+    return data
+
+
 def flatten(data, nested_columns=None):
     nested_columns = set(nested_columns or NESTED_COLUMNS)
     if not nested_columns.issubset(NESTED_COLUMNS):
         raise NestedColumnError(nested_columns)
-    if not data:
-        return data
-    if isinstance(data, list):
-        keys = set(data[0].keys()) & nested_columns
-        for item in data:
-            for key in keys:
-                item.update({f"{key}_{k}": v for k, v in item.get(key).items()})
-                item.pop(key)
+    if isinstance(data, DataFrame) and not data.empty:
+        keys = set(data.columns) & nested_columns
+        data = data.to_dict(orient="records")
```

Comment:
> I already have change de object type.

The change I commented on didn't make a lot of sense. Why not get rid of it instead? So:
* better performance (less needed transformation)
* more data type safety (Pandas DataFrame types differ from Python dictionaries)
* Avoid something that Python's official doc recommends avoiding

Reply:
> better performance (less needed transformation)
I am about to commit a change I ahve done. But somehow I still using dict...

 > more data type safety (Pandas DataFrame types differ from Python dictionaries)
How could I improve this point?

> Avoid something that Python's official doc recommends avoiding
What exacltly I should avoid? using data type to chenck object class?

Reply:
> How could I improve this point?

Ifd the input is a DataFrame _and_ the output is also a DataFrame, do not convert it to a dictionary in the middle of the function.

> What exacltly I should avoid?

[Using `type(blablabla) == DataFrame`](https://github.com/FelipeSBarros/crossfire/pull/103#discussion_r1501697218).

Reply:
Ok. I am not using it anymore.